docroot/modules/custom/mass_utility/src/Commands/MassUtilityCommands.php

```diff
@@ -398,4 +399,95 @@ public function queueRevisionsCleanup(array $options = [
 
   }
 
+  /**
+   * Copy D2D nodes into redirect entities.
+   *
+   * @command copyd2d
+   * @option limit
+   * @option min-hits
+   */
+  public function copyD2DRedirects($options = ['limit' => self::REQ, 'min-hits' => 50]) {
+    $query = $this->database->select('node__field_legacy_redirects_legacyurl', 'source')
+      ->fields('source', ['entity_id', 'delta']);
+    $query->addField('source', 'field_legacy_redirects_legacyurl_value', 'source');
+    $query->addField('target', 'field_legacy_redirects_ref_conte_target_id', 'target');
+    if ($options['limit']) {
+      $query->range(0, $options['limit']);
+    }
+    $query->innerJoin('node__field_legacy_redirects_ref_conte', 'target', 'source.entity_id = target.entity_id AND source.delta = target.delta');
+    $records = $query->execute()->fetchAll();
```

Comment:
Before we execute... can we limit this by `status` = 1 and `field_legacy_redirect_env_value` = 0?